javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -164,14 +164,21 @@ private Vector(BitMappedTrie<T> trie) {
         if (elements instanceof Vector) {
             return (Vector<T>) elements;
         } else {
-            BitMappedTrie<T> trie = BitMappedTrie.empty();
-            for (T element : elements) {
-                trie = trie.append(element);
-            }
-            return ofAll(trie);
+            final Seq<? extends T> seq = seq(elements);
+            return ofAll(seq, seq.size());
         }
     }
 
+    public static <T> Vector<T> ofAll(Collection<? extends T> elements) {
+        Objects.requireNonNull(elements, "elements is null");
+        return ofAll(elements, elements.size());
+    }
+
+    public static <T> Vector<T> ofAll(Iterable<? extends T> iterable, int size) {
```

Comment:
Why do we need this public method? The user should not be bothered providing a size, right? If it is needed internally I would reduce the visibility.